### DISCUSSION TITLE
Lesson 4 Can't print(transaction)

### DISCUSSION THREAD

**Discussion by chioke07** [smartcontractkit/full-blockchain-solidity-course-py#1853] [General]
<img width="1512" alt="Screenshot 2023-03-30 at 3 59 57 AM" src="https://user-images.githubusercontent.com/129320402/228694468-17a18580-f0f8-4c0a-95d7-a646ccfdb5db.png">

i think I can't connect to ganache, i don't know why but i followed the video. I'm confused, this is the error I'm getting, please how can i fix it? 

requests.exceptions.ConnectionError: HTTPConnectionPool(host='0.0.0.0', port=8545): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x105764e50>: Failed to establish a new connection: [Errno 61] Connection refused'))

**Comment by Niferu** [comment]
Hello @chioke07 

Are you running this code with opened `Ganache`? Have you checked if your private key and provider are correct and match Ganache ones?

I can already see there is something wrong with your `HTTPProvider` address as it should look like `127.0.0.1:8545` and also `chainId` should be `1337` not `5777`
